### PR TITLE
manifest + static analysis

### PR DESCRIPTION
**Pull request by McCallisterRomer**
- updated manifest to json format
- added /analyzer option for msvc
  - renamed `WARNING_FLAGS` CMake variable, becuase its not just warnings.
  - external directory included as SYSTEM to avoid generating warnings for them
- updated docs
- deleted last of the original nctools.exe stuff

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,35 +8,52 @@ endif()
 set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
+option(NC_TOOLS_BUILD_CONVERTER "Build nc-convert executable" ON)
+option(NC_TOOLS_BUILD_TESTS "Include tests in build" ON)
+option(NC_TOOLS_STATIC_ANALYSIS "Enable static analysis (MSVC Only" OFF)
+
+set(NC_TOOLS_COMPILE_OPTIONS
```
Comment (McCallisterRomer) [inline CMakeLists.txt]: delete

File: CMakeLists.txt [hunk 2]
```diff
@@ -8,35 +8,52 @@ endif()
...
+        -Wpedantic
+        -Wshadow
+        -Wconversion
+        -Wfatal-errors
+        -Wno-cast-function-type
+    )
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    set(WARNING_FLAGS /W3 /WX)
+    set(NC_TOOLS_COMPILE_OPTIONS
+        /W3
```
Comment (McCallisterRomer) [inline CMakeLists.txt]: I feel like I tried /W4 here, but can't remember why I went back. Worth trying agin.

File: CMakeLists.txt [hunk 3]
```diff
@@ -8,35 +8,48 @@ endif()
 set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
+option(NC_TOOLS_BUILD_CONVERTER "Build nc-convert executable" ON)
+option(NC_TOOLS_BUILD_TESTS "Include tests in build" ON)
+option(NC_TOOLS_STATIC_ANALYSIS "Enable static analysis (MSVC Only" OFF)
```
Comment (jaremieromer) [inline CMakeLists.txt]: Missing a ')' in "(MSVC Only"

File: README.md [hunk 4]
```diff
@@ -2,4 +2,81 @@
...
+        }
+    ]
+}
+```
+
+```
+> nc-convert -m manifest.json
+```
+
+`nc-convert` will skipt files that are already up-to-date when using a manifest.
```
Comment (jaremieromer) [inline README.md]: typo: "skipt"

File: README.md [hunk 5]
```diff
@@ -2,4 +2,81 @@
...
+
+For more information, see the help text for `nc-convert` and the docs on [input file
+requirements](docs/SourceFileRequirements.md) and [.nca formats](docs/AssetFormats.md)
+
+[NcCommon]() is a public dependency of NcAsset. The following third-party libraries are used internally by nc-convert: [Assimp](), [AudioFile](), and [stb_image]().
```
Comment (jaremieromer) [inline README.md]: These links do not seem to work

File: docs/SourceFileRequirements.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+Source File Requirements
+========================
+
+## Audio Conversion
+> Supported file types: wav
```
Comment (jaremieromer) [inline docs/SourceFileRequirements.md]: "wav" should be ".wav" to match the format of the other file types

File: source/ncconvert/Main.cpp [hunk 7]
```diff
@@ -8,17 +8,58 @@
 #include <iostream>
 #include <string>
 
-const auto DefaultAssetTargetFilename = std::string{"targets.txt"};
+constexpr auto usageMessage = 
+R"(Usage: nc-convert [options]
+Options:
```
Comment (jaremieromer) [inline source/ncconvert/Main.cpp]: Remove this colon to be consistent with the other headers

File: source/ncconvert/Main.cpp [hunk 8]
```diff
@@ -32,96 +73,72 @@ int main(int argc, char** argv)
...
 
-    out->outputDirectory = std::filesystem::path("./");
     auto option = std::string{};
     auto current = 1;
     while (current < argc)
     {
         option = argv[current];
 
-        if(option.compare("-h") == 0 || option.compare("--help") == 0)
+        if (option == "-h")
```
Comment (jaremieromer) [inline source/ncconvert/Main.cpp:112]: it would be nice to do a std::tolower on option so that it is case insensitive

**Review by jaremieromer — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (jaremieromer) [reply README.md]: (Assimp, AudioFile, stb_image)

File: README.md (hunk 4 above)
Comment (McCallisterRomer) [reply README.md]: skipt-o my lou

File: source/ncconvert/Config.h [hunk 9]
```diff
@@ -12,9 +12,31 @@ namespace nc::convert
...
-    std::optional<std::filesystem::path> singleTargetPath;
-    std::optional<asset::AssetType> singleTargetType;
-    std::optional<std::filesystem::path> targetsFilePath;
-    std::filesystem::path outputDirectory;
+    /** @brief The directory to output .nca files to. */
+    std::filesystem::path outputDirectory = "./";
+
+    /**
+     * @brief A path to a raw asset file to be converted to .nca format.
+     * @note Specific to Mode::Target
```
Comment (McCallisterRomer) [inline source/ncconvert/Config.h]: I think I deleted the `Mode` enum. Reword these.

File: source/ncconvert/Main.cpp [hunk 10]
```diff
@@ -8,17 +8,58 @@
...
+Json Manifest
+  A provided manifest should be a json file containing an 'assets' array with
+  one or more conversion specifications, and an optional 'globalOptions' object
+  defining global settings. Example:
+  {
+      "globalOptions": {
+          "outputDirectory": "outDir", // default: "./"
+          "workingDirectory": "./"     // default: "./"
+      },
+      "assets": [
```
Comment (McCallisterRomer) [inline source/ncconvert/Main.cpp]: Update this with the per-asset arrays.

**Pull request merged**